frontend/src/Components/ContentComponent.tsx

```diff
@@ -651,6 +651,38 @@ function ZoomComponent(props: ZoomComponentProps) {
     const defaultTranslateY = (window.innerHeight - props.height * defaultScale) / 2;
     useHotkeys('esc', props.onExit);
 
+    useEffect(() => {
+
+        // Event listener to handle back navigation
+        const handleBack = (event: PopStateEvent) => {
+            // preventing default back action if the new state isn't the popup
+            if (!event.state.popupOpen) {
+                // Preventing the default back action
+                event.preventDefault();
+                // Call the function to close the overlay
+                props.onExit();
+            }
+        };
+
+        // make sure we're only adding the event listener once
+        if (!window.history.state.popupOpen) {
+            // Pushing a new entry onto the history stack
+            window.history.pushState({popupOpen: true}, '');
+            // Adding event listener
+            window.addEventListener('popstate', handleBack);
+        }
+
+        // Cleanup function to restore the state
+        return () => {
+            // Removing event listener
+            window.removeEventListener('popstate', handleBack);
+            // Try to move back in history stack if the popup was open
+            if (window.history.state && window.history.state.popupOpen) {
+                window.history.back();
```

Comment:
I think this will leave "popupOpen" state in the history stack ("forward" button will be enabled on toolbar on desktop).

Reply:
Yep, that's part of the bug that I described.